misk-hibernate/src/main/kotlin/misk/hibernate/JdbcExtensions.kt

```diff
@@ -0,0 +1,15 @@
+package misk.hibernate
+
+import com.google.common.collect.Iterables.getOnlyElement
+import java.sql.ResultSet
+
+fun <T> ResultSet.map(function: (ResultSet) -> T): List<T> {
+  val result = mutableListOf<T>()
+  while (this.next()) {
+    result.add(function(this))
+  }
+  return result
+}
+
+fun <T> ResultSet.single(function: (ResultSet) -> T): T = getOnlyElement(map(function))
```

Comment:
same

Reply:
Also – elsewhere we have a method uniqueResult() that returns a `T?` . . . I think this should either be `internal` or be symmetric with that method

Reply:
Thanks! I was trying to find a precedent to be symmetrical with. :-)